var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -159,6 +159,8 @@ class Petsc(Package):
 
     patch('xlf_fix-dup-petscfecreate.patch', when='@3.11.0')
 
+    depends_on('diffutils', type='build')
```

Comment:
`diff` is primarily used by the test suite scripts [and configure set it up for that purpose]. So perhaps its a `type='run` dependency? [if we were to add a test section to the petsc spack spec]

I don't know if it matters - as once diff is installed - its available for use at runtime anyway [as configure grabs and stores this location for future use].

In some sense - this dependency is similar to mpiexec.

Reply:
Ok. I didn't know that. I was seeing the error during the configure phase of build. Should we say it is both build and run?

Reply:
Its likely  run only  [configure checks for it similar to other run dependencies like MPI, blas etc for future use] - and does not use it during build of petsc library. But I guess its fine to list both.

I get a bit lost between the subtle differences between build and run dependencies. And then there are library dependencies [-lblas] and tool dependencies [mpiexec] - which are not exactly the same..

Reply:
I just confirmed that IF a dependency is needed during *any* phase of the install, including if needed for `configure` phase, it has to be listed as a "build" dependency.

Reply:
I guess part of the issue for me is: if we add a `test()` section - are its dependencies build dependencies or run dependencies or both. I don't know. 

[this section does compiles  of test binaries - aka build deps - and run them using scripts - aka run deps?]

Reply:
> I just confirmed that IF a dependency is needed during any phase of the install, including if needed for configure phase, it has to be listed as a "build" dependency.

Ok in this sense all petsc dependencies are both build and run dependencies.

There is a dependency to run configure script [python] - and when configure is run - it checks for all petsc dependences - sometimes runs them to check if the work [so for ex: it compiles and runs mpi binaries etc, checks for all link dependencies and tool dependencies, some of them are used for build - for eg compilers, ar etc, and some are used for test suite - diff, compilers etc..]